### ISSUE TITLE
Verify if latest code or specific commit from Kitura-Build is getting pulled when building locally.

### ISSUE THREAD

**Issue by rolivieri** [IBM-Swift/Kitura-Build#12]

**Comment by ianpartridge** [comment]
From the Kitura repo:

<img width="259" alt="screen shot 2016-08-01 at 15 56 15" src="https://cloud.githubusercontent.com/assets/1423497/17298108/86b8b8cc-5800-11e6-9dac-228e232cf5ba.png">

This shows that a specific (old) commit hash of Kitura-Build is being used.

**Comment by dfirsht** [comment]
Part of the travis process is to check out the latest master of Kitura-Build 
<img width="1051" alt="screen shot 2016-08-01 at 9 57 51 am" src="https://cloud.githubusercontent.com/assets/5191832/17298182/8b0f44a4-57ce-11e6-872b-1759a0701af3.png">

**Comment by ianpartridge** [comment]
Oh so it doesn't matter that Kitura is referring to an old commit in Kitura-Build?

**Comment by dfirsht** [comment]
Not on Travis, local builds though are different. This code is in the `.travis.yml`, we could move it to the Makefile if we wanted it to always happen.

**Comment by ianpartridge** [comment]
@rolivieri what do you think?

**Comment by shmuelk** [comment]
Travis does a git clone --recursive when cloning the repository to be tested. That will pull Kitura-Build based on the pushed commit reference. The Makefile will not upgrade an existing Kitura-Build directory.

**Comment by dfirsht** [comment]
@shmuelk As my screenshot shows, Travis then afterwards does a `git checkout master` in Kitura-Build to pull the latest off master. I was suggesting that we move this code to the Makefile so this will happen in local builds as well. Note that this will have to be in the external Makefiles in the repos, not the master Makefile in Kitura-Build.

**Comment by shmuelk** [comment]
But only if the git clone --recursive works

**Comment by rolivieri** [comment]
The makefiles may have to be updated as @dfirsht mentioned above. @BDHernand is looking into this issue and will provide guidance once he completes testing.

**Comment by BDHernand** [comment]
Currently in the makefiles we are using 

git submodule update --init

This is equivalent to doing

git submodule init
git submodule update

What it is not doing is makeing sure you are getting the latest from the main branch.  When I diffed the Kitura-Build directory with the repos it was not at the latest code.  I changed the command to be the following:

git submodule update --init --remote --recursvie

--remote causes the remote branch that is used to default to master
--recursive will cause all submodules to be updated recursively

There is an implicit --checkout on the command so we dont have to do a checkout master since we did the --remote.

I have tested the build locally with these changes and without the --remote --recursive we can not guarantee we are getting the latest.  With them the changes i have pushed into Kitura-Build were present
